cmd/kubeadm/app/apis/kubeadm/v1beta4/conversion.go

```diff
@@ -53,6 +53,7 @@ func Convert_v1beta4_ClusterConfiguration_To_kubeadm_ClusterConfiguration(in *Cl
 // Convert_v1beta4_JoinConfiguration_To_kubeadm_JoinConfiguration converts a public JoinConfiguration to a private JoinConfiguration.
 func Convert_v1beta4_JoinConfiguration_To_kubeadm_JoinConfiguration(in *JoinConfiguration, out *kubeadm.JoinConfiguration, s conversion.Scope) error {
 	err := autoConvert_v1beta4_JoinConfiguration_To_kubeadm_JoinConfiguration(in, out, s)
+	out.Discovery.Timeout = in.Timeouts.Discovery.DeepCopy()
```

Comment:
looks like in.Timeouts can be nil here?

Reply:
in.Timeouts is from the public API as a pointer field with omitempty, but the defaulter before the converter creates the struct if it's nil, which in a way defeats the omitempty aspect if that same config is then serialized:
https://github.com/kubernetes/kubernetes/blob/bb2068b19bddad0305126543bd8cdf455db53b2e/cmd/kubeadm/app/apis/kubeadm/v1beta4/defaults.go#L146-L149

Reply:
ok... does everything that comes through conversion have defaulting applied to it?